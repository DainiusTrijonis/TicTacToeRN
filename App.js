import React, {Component} from 'react';
import { StyleSheet, Text, View, TouchableOpacity,Button, Alert } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';



export default class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      gameState: [
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]
      ],
        currentPlayer: this.getStartingPlayer(),
        gameType:3  
    }
  }

  initializeGame = () => {
    this.setState({gameState:
      [
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]
      ]
    });
  }

  generateRandomNumber() {
    var RandomNumber = Math.floor(Math.random() * 10) +1;
    return RandomNumber;
  }
  getStartingPlayer(){
    var randomNumber = this.generateRandomNumber()
    if(randomNumber >= 5)
    {
      return 1;
    }
    else
    {
      return -1;
    }

  }
  componentDidMount() {
    this.initializeGame();
  }


  generateRandomNumber = () => {
    var RandomNumber = Math.floor(Math.random() * 10) +1;
    return RandomNumber;
  }


  setGameMode = (mode) =>{
    this.state.gameType = mode;
    this.initializeGame();
  }

  onTilePress = (row,col) => {

    if(this.state.gameState[row][col] != 0) { return; }

    var currentPlayer = this.state.currentPlayer;

    // Set the corret tile
    var arr = this.state.gameState.slice();
    arr[row][col] =currentPlayer;
    this.setState({gameState:arr});
    
    //Switch to other player
    var nextPlayer = (currentPlayer == 1) ? -1 : 1;
    this.setState({currentPlayer: nextPlayer});
    

    var winner = this.getWinner();
    if(winner == 1)
    {
      Alert.alert("Player 1 is the winner");
      this.initializeGame();
    }
    else if(winner == -1)
    {
      Alert.alert("Player 2 is the winner");
      this.initializeGame();
    }

  }

  getWinner = () =>{
    const Num_Tiles = this.state.gameType;
    var arr = this.state.gameState;

    var sum=0;
    //check rows

    for(var i=0; i<Num_Tiles; i++)
    {
      sum=0;
      for(var j=0; j<Num_Tiles; j++)
      {
        sum+= arr[i][j];
      }
      if (sum == Num_Tiles){return 1}
      else if( sum == Num_Tiles*-1) { return -1}

    }

    //check columns

    for(var j=0; j<Num_Tiles; j++)
    {
      sum=0;
      for(var i=0; i<Num_Tiles; i++)
      {
        sum+= arr[i][j];
      }
      if (sum == Num_Tiles){return 1}
      else if( sum == Num_Tiles*-1) { return -1}
    }

    //check diagonals
 
    sum=0;
    for(var i=0; i<Num_Tiles; i++)
    {
        sum+=arr[i][i];
    }
    if (sum == Num_Tiles){return 1}
    else if( sum == Num_Tiles*-1) { return -1}

    sum=0;
    var j=0;
    for(var i=Num_Tiles-1; i>=0; i--)
    {

      sum+=arr[i][j];//0,2; 1,1; 2,0;
      j++;

    }
    if (sum == Num_Tiles){return 1}
    else if( sum == Num_Tiles*-1) { return -1}
  }



  renderIcon = (row,col) => {
    var value = this.state.gameState[row][col];
    switch(value)
    {
      case 1: return <Icon name="x" size={30} color="green" />;
      case -1: return <Icon name="circle" size={30} color="red" />;
      default : return <View/>;
    }
  }
  render() {
    var gameType = this.state.gameType;
    switch(gameType)
    {
      case 3:   return(
        <View style={styles.container}>
          <Button title="Game Type 4" onPress={()=> this.setGameMode(4)}/>
          <Button title="Game Type 5" onPress={()=> this.setGameMode(5)}/>
        <View style= {{flexDirection: 'row'}}>
            <TouchableOpacity onPress={() => this.onTilePress(0,0)} style={styles.tile}>
              {this.renderIcon(0,0)}
            </TouchableOpacity>
            
            <TouchableOpacity onPress={() => this.onTilePress(0,1)} style={styles.tile}>
              {this.renderIcon(0,1)}
            </TouchableOpacity>

            <TouchableOpacity onPress={() => this.onTilePress(0,2)} style={styles.tile}>
              {this.renderIcon(0,2)}
            </TouchableOpacity>
        </View>


        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(1,0)} style={styles.tile}>
            {this.renderIcon(1,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,1)} style={styles.tile}>
            {this.renderIcon(1,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,2)} style={styles.tile}>
            {this.renderIcon(1,2)}
          </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(2,0)} style={styles.tile}>
            {this.renderIcon(2,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,1)} style={styles.tile}>
            {this.renderIcon(2,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,2)} style={styles.tile}>
            {this.renderIcon(2,2)}
          </TouchableOpacity>
        </View>

        <Button title="New Game" onPress={()=> this.initializeGame()}/>

      </View>
      );
      case 4:    return(
        <View style={styles.container}>
          <Button title="Game Type 3" onPress={()=> this.setGameMode(3)}/>
          <Button title="Game Type 5" onPress={()=> this.setGameMode(5)}/>
        <View style= {{flexDirection: 'row'}}>
            <TouchableOpacity onPress={() => this.onTilePress(0,0)} style={styles.tile}>
              {this.renderIcon(0,0)}
            </TouchableOpacity>
            
            <TouchableOpacity onPress={() => this.onTilePress(0,1)} style={styles.tile}>
              {this.renderIcon(0,1)}
            </TouchableOpacity>

            <TouchableOpacity onPress={() => this.onTilePress(0,2)} style={styles.tile}>
              {this.renderIcon(0,2)}
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.onTilePress(0,3)} style={styles.tile}>
              {this.renderIcon(0,3)}
            </TouchableOpacity>
        </View>


        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(1,0)} style={styles.tile}>
            {this.renderIcon(1,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,1)} style={styles.tile}>
            {this.renderIcon(1,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,2)} style={styles.tile}>
            {this.renderIcon(1,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,3)} style={styles.tile}>
              {this.renderIcon(1,3)}
            </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(2,0)} style={styles.tile}>
            {this.renderIcon(2,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,1)} style={styles.tile}>
            {this.renderIcon(2,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,2)} style={styles.tile}>
            {this.renderIcon(2,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,3)} style={styles.tile}>
              {this.renderIcon(2,3)}
            </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(3,0)} style={styles.tile}>
            {this.renderIcon(3,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,1)} style={styles.tile}>
            {this.renderIcon(3,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,2)} style={styles.tile}>
            {this.renderIcon(3,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,3)} style={styles.tile}>
              {this.renderIcon(3,3)}
            </TouchableOpacity>
        </View>

        <Button title="New Game" onPress={()=> this.initializeGame()}/>

      </View>

      );
      case 5: return(
        <View style={styles.container}>
          <Button title="Game Type 3" onPress={()=> this.setGameMode(3)}/>
          <Button title="Game Type 4" onPress={()=> this.setGameMode(4)}/>
        <View style= {{flexDirection: 'row'}}>
            <TouchableOpacity onPress={() => this.onTilePress(0,0)} style={styles.tile}>
              {this.renderIcon(0,0)}
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.onTilePress(0,1)} style={styles.tile}>
              {this.renderIcon(0,1)}
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.onTilePress(0,2)} style={styles.tile}>
              {this.renderIcon(0,2)}
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.onTilePress(0,3)} style={styles.tile}>
              {this.renderIcon(0,3)}
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.onTilePress(0,4)} style={styles.tile}>
              {this.renderIcon(0,4)}
            </TouchableOpacity>
        </View>


        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(1,0)} style={styles.tile}>
            {this.renderIcon(1,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,1)} style={styles.tile}>
            {this.renderIcon(1,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,2)} style={styles.tile}>
            {this.renderIcon(1,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,3)} style={styles.tile}>
            {this.renderIcon(1,3)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(1,4)} style={styles.tile}>
            {this.renderIcon(1,4)}
          </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(2,0)} style={styles.tile}>
            {this.renderIcon(2,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,1)} style={styles.tile}>
            {this.renderIcon(2,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,2)} style={styles.tile}>
            {this.renderIcon(2,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,3)} style={styles.tile}>
              {this.renderIcon(2,3)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(2,4)} style={styles.tile}>
              {this.renderIcon(2,4)}
          </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(3,0)} style={styles.tile}>
            {this.renderIcon(3,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,1)} style={styles.tile}>
            {this.renderIcon(3,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,2)} style={styles.tile}>
            {this.renderIcon(3,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,3)} style={styles.tile}>
              {this.renderIcon(3,3)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(3,4)} style={styles.tile}>
              {this.renderIcon(3,4)}
          </TouchableOpacity>
        </View>
        <View style= {{flexDirection: 'row'}}>
          <TouchableOpacity onPress={() => this.onTilePress(4,0)} style={styles.tile}>
            {this.renderIcon(4,0)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(4,1)} style={styles.tile}>
            {this.renderIcon(4,1)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(4,2)} style={styles.tile}>
            {this.renderIcon(4,2)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(4,3)} style={styles.tile}>
              {this.renderIcon(4,3)}
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.onTilePress(4,4)} style={styles.tile}>
              {this.renderIcon(4,4)}
          </TouchableOpacity>
        </View>

        <Button title="New Game" onPress={()=> this.initializeGame()}/>

      </View>
      );

    }

    return <View><Button title = "Failed rendering "/></View>;

  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tile:{
    borderWidth: 1,
    width:80,
    height:80,
    alignItems: 'center',
    justifyContent: 'center',
  },
  tileX:{
    color:'red',
    fontSize:50,

  },
  tileO:{
    color:'green',
    fontSize:50,
  },
});
