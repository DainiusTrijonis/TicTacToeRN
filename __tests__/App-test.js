/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<App rendering />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has 6 child when gameType is 3 tiles', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(6);
  });
});
describe('<GameLogic />', () => {

  describe('<GameLogic of GameType 3 tiles />', () => {
    let AppData = renderer.create(<App />).getInstance();
    AppData.setGameMode(3);
    it('Get a win on diagonal with player 1', () => {
      var arr =
      [
        [1, 0, 0, 0, 0],
        [0, 1, 0, 0, 0],
        [0, 0, 1, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(1);
    });
    it('Get a win on diagonal with player 2', () => {
      var arr =
      [
        [-1, 0, 0, 0, 0],
        [0, -1, 0, 0, 0],
        [0, 0, -1, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(-1);
    });
    it('Get a win on Column with player 1', () => {
        var arr =
        [
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Column with player 2', () => {
        var arr =
        [
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
      it('Get a win on Line with player 1', () => {
        var arr =
        [
          [1, 1, 1, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Line with player 1', () => {
        var arr =
        [
          [-1, -1, -1, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
  });
  describe('<GameLogic of GameType 4 tiles />', () => {
    let AppData = renderer.create(<App />).getInstance();
    AppData.setGameMode(4);
    it('Get a win on diagonal with player 1', () => {
      var arr =
      [
        [1, 0, 0, 0, 0],
        [0, 1, 0, 0, 0],
        [0, 0, 1, 0, 0],
        [0, 0, 0, 1, 0],
        [0, 0, 0, 0, 0]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(1);
    });
    it('Get a win on diagonal with player 2', () => {
      var arr =
      [
        [-1, 0, 0, 0, 0],
        [0, -1, 0, 0, 0],
        [0, 0, -1, 0, 0],
        [0, 0, 0, -1, 0],
        [0, 0, 0, 0, 0]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(-1);
    });
    it('Get a win on Column with player 1', () => {
        var arr =
        [
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Column with player 2', () => {
        var arr =
        [
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
      it('Get a win on Line with player 1', () => {
        var arr =
        [
          [1, 1, 1, 1, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Line with player 2', () => {
        var arr =
        [
          [-1, -1, -1, -1, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
  });
  describe('<GameLogic of GameType 5 tiles />', () => {
    let AppData = renderer.create(<App />).getInstance();
    AppData.setGameMode(5);
    it('Get a win on diagonal with player 1', () => {
      var arr =
      [
        [1, 0, 0, 0, 0],
        [0, 1, 0, 0, 0],
        [0, 0, 1, 0, 0],
        [0, 0, 0, 1, 0],
        [0, 0, 0, 0, 1]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(1);
    });
    it('Get a win on diagonal with player 2', () => {
      var arr =
      [
        [-1, 0, 0, 0, 0],
        [0, -1, 0, 0, 0],
        [0, 0, -1, 0, 0],
        [0, 0, 0, -1, 0],
        [0, 0, 0, 0, -1]
      ]
      AppData.setState({gameState:arr});
      var winner = AppData.getWinner();
      expect(winner).toEqual(-1);
    });
    it('Get a win on Column with player 1', () => {
        var arr =
        [
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0],
          [1, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Column with player 2', () => {
        var arr =
        [
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0],
          [-1, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
      it('Get a win on Line with player 1', () => {
        var arr =
        [
          [1, 1, 1, 1, 1],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(1);
      });
      it('Get a win on Line with player 2', () => {
        var arr =
        [
          [-1, -1, -1, -1, -1],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0],
          [0, 0, 0, 0, 0]
        ]
        AppData.setState({gameState:arr});
        var winner = AppData.getWinner();
        expect(winner).toEqual(-1);
      });
  });
  describe('<Gamelogic other functions />', () => {
    let AppData = renderer.create(<App />).getInstance();
    it('random starting player not to be 0', () => {

      var startingPlayer = AppData.getStartingPlayer();
      expect(startingPlayer).not.toBe(0);
    });
    it('Check if initializing clears game array ', () => {
      var changedArray =
      [
        [1, 0, 0, 0, 0],
        [0, 1, 0, 0, 0],
        [0, 0, 1, 0, 0],
        [0, 0, 0, 1, 0],
        [0, 0, 0, 0, 1]
      ]
      var clearArray =
      [
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0]
      ]
      AppData.setState({gameState:changedArray});

      AppData.initializeGame();

      expect(AppData.state.gameState).toEqual(clearArray);
    });
  });
});

