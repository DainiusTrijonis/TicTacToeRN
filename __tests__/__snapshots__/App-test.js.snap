// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App rendering /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Game Type 4
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Game Type 5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        New Game
      </Text>
    </View>
  </View>
</View>
`;
